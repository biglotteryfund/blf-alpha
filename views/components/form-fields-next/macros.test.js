/* eslint-env jest */
'use strict';
const { renderComponentMacro } = require('../../../modules/test-renderers');

describe.only('form field macros', () => {
    test('inputText', async () => {
        const result = await renderComponentMacro('components/form-fields-next/macros.njk', 'inputText', {
            name: 'example',
            type: 'text',
            label: 'First name',
            isRequired: true
        });
        expect(result).toMatchSnapshot();
    });

    test('inputCurrency', async () => {
        const result = await renderComponentMacro('components/form-fields-next/macros.njk', 'inputCurrency', {
            name: 'example',
            label: 'Amount',
            isRequired: true
        });
        expect(result).toMatchSnapshot();
    });
});
