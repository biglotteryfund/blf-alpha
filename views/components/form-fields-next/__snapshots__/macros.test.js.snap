// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form field macros inputCurrency 1`] = `

<label class="ff-label"
       for="field-example"
>
  Amount
</label>
<div class="ff-currency ff-currency--constrained">
  <div class="ff-currency__pre">
    Â£
  </div>
  <input class="ff-currency__input"
         type="number"
         id="field-example"
         name="example"
         required
         aria-required="true"
  >
</div>

`;

exports[`form field macros inputText 1`] = `

<label class="ff-label"
       for="field-example"
>
  First name
</label>
<input class="ff-text"
       type="text"
       id="field-example"
       name="example"
       required
       aria-required="true"
       size="40"
>

`;
